text/0000-deprecate-anonymous-parameters.md

```diff
@@ -0,0 +1,205 @@
+- Feature Name: deprecate_anonymous_parameters
+- Start Date: 2016-07-19
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Currently Rust allows anonymous parameters in trait methods:
+
+```Rust
+trait T {
+    fn foo(i32);
+
+    fn bar_with_default_impl(String, String) {
+
+    }
+}
+```
+
+This RFC proposes to deprecate this syntax. This RFC intentionally does not
+propose to remove this syntax.
+
+# Motivation
+[motivation]: #motivation
+
+Anonymous parameters are a [historic accident]. They cause a number of technical
```

Comment:
I was referring to another part of the comment, but maybe I've misinterpreted it

> I thought we planned to require parameter names in trait definitions for this reason -- but I guess that never happened?

@nikomatsakis what would be the right way to phrase this in the RFC? Or we can just drop this sentence: it does not add anything to the current situation, it just an interesting historical detail :)